src/_pytest/stepwise.py

```diff
@@ -48,6 +48,8 @@ def pytest_configure(config: Config) -> None:
 def pytest_sessionfinish(session: Session) -> None:
     if not session.config.getoption("stepwise"):
         assert session.config.cache is not None
+        if hasattr(session.config, "workerinput"):
```

Comment:
```suggestion
        # Do not update cache if this process is a xdist worker to prevent race conditions (#10641).
        if hasattr(session.config, "workerinput"):
```

Reply:
Your feedback has been incorporated into the code